### ISSUE TITLE
Better support for fork/exec

### ISSUE THREAD

**Issue by rschoene** [tud-zih-energy/lo2s#143]
I'm currently seeing a lot of wrongly mapped pids (mapped to the wrong executable) when using `lo2s mpirun` on a system. Some processes are mapped correctly, some processes are called like their parent process.
I assume that this is due to the mapping being done at the beginning of tracing a process. Depending on whether this is done before or after the the exec call, it gets either the old name and mapping or the new one.
It'd be good if that mapping would be (1) either done after exec (i.e. at the ending of a process) or (2) the mapping could be re-evaluated at an exec (or when it sees that the mapping of memory region changes accordingly).

Version: v1.2.2-40-gf5dbe26
started with: `lo2s -X -e cpu-cycles -E instructions -E cpu-cycles -- mpirun -np 24 bin/lu.C.x`

**Comment by bmario** [comment]
I created several traces, but none of them looks odd. Can you attach the trace file? And what version of Vampir are you using?

**Comment by rschoene** [comment]
Here you go ...
[lo2s_143_fork_exec.zip](https://github.com/tud-zih-energy/lo2s/files/4138686/lo2s_143_fork_exec.zip)

**Comment by cvonelm** [comment]
I have tested this with the following program:

```c
//test1.c
workload();
pid = fork();
if(!pid)
{
exec("test2");
}
workload();
```

```c
//test2.c
workload();
pid = fork();
if(!pid)
{
exec("test3");
}
workload();
```
```c
//test3.c
workload();
```
test1 and test2 are always matched correctly, but test3 is always named test2

**Comment by cvonelm** [comment]
I have written a small fork-exec chain generator which can be found here: https://github.com/cvonelm/forkbench

With a 10 seconds sleep between the fork and the exec, we are not missing the PERF_RECORD_COMM events that indicate that the process name has changed

Without that delay we begin to loose PERF_RECORD_COMM events again.

So it seems that we got a timing problem on our hands

**Comment by cvonelm** [comment]
TL;DR: No new leads, just documenting what I've found out till now:

**Buffer size**
Increasing or decreasing the mmap buffer size has no effect on the amount of comm events lo2s misses.

**The event set-up process**
When a process is forked off off an existing process that lo2s monitors via ptrace
the original and forked processes are stopped and lo2s is signaled. lo2s then sets up and _enables_ the events (including the comm events) before giving back control to the monitored application. In reality this works too, whether we miss the comm event or not, the counters are always enabled before the exec happens.
